src/confirm_role/pages/ConfirmRole.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { BeatLoader } from 'react-spinners';
+import { parse } from 'querystring';
+import { Heading, Paragraph } from '../../shared/components/text/base';
+import { FlexContainerCol } from '../../shared/components/layout/base';
+import NavHeader from '../../shared/components/NavHeader';
+import { Visitors, ResponseUtils, ErrorUtils } from '../../api';
+import { colors } from '../../shared/style_guide';
+
+
+const StyledSection = styled.section`
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+  text-align: center;
+`;
+
+const ErrorText = styled(Paragraph)`
+  color: ${colors.error};
+`;
+
+const status = {
+  PENDING: 'PENDING',
+  SUCCESS: 'SUCCESS',
+  REDIRECT: 'REDIRECT',
+  FAILURE: 'FAILURE',
+};
+
+export default class ConfirmRole extends Component {
+  constructor(props) {
+    super(props);
+
+    const { props: { location: { search } } } = this;
+    const { role } = parse(search.replace('?', ''));
+
+    this.state = {
+      status: status.PENDING,
+      role,
+    };
+  }
+
+  componentDidMount() {
+    const { props: { location: { search }, match: { params: { token } } } } = this;
+    const { userId, organisationId, role } = parse(search.replace('?', ''));
+    Visitors.addRole({ token, userId, organisationId, role: role && role.toUpperCase() })
+      .then((r) => {
+        const res = ResponseUtils.getResponse(r);
+        return res.token
+          ? this.setState({ status: status.REDIRECT, user: res })
```

Comment:
Is this case possible? Looking at [`/register/confirm`](https://github.com/TwinePlatform/twine-api/pull/371/files#diff-3f19b613b2c353cbf665d7d878b06b61), it looks like you only get a token if you are registering a VOLUNTEER role to a pre-existing VISITOR account, which can't happen on the visitor app, no?

Reply:
Both emails from the visitor app and volunteer app redirect to this page.